### PR TITLE
Handle Hive tables with bucket property partially set

### PR DESCRIPTION
**Pull request by haozhun**
In Hive, a table is considered as not bucketed when its bucketCols is set but
its numBucket is not set. Previously, Presto consider such table as corrupted.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java [hunk 1]
```diff
@@ -41,11 +42,15 @@ public HiveBucketProperty(@JsonProperty("clusteredBy") List<String> clusteredBy,
     {
         boolean bucketColsSet = storageDescriptor.isSetBucketCols() && !storageDescriptor.getBucketCols().isEmpty();
         boolean numBucketsSet = storageDescriptor.isSetNumBuckets() && storageDescriptor.getNumBuckets() > 0;
-        if (bucketColsSet != numBucketsSet) {
-            throw new PrestoException(HiveErrorCode.HIVE_INVALID_METADATA, "Only one of bucketCols and numBuckets is set in metadata of table/partition " + tablePartitionName);
+        if (!numBucketsSet) {
+            // In Hive, a table is considered as not bucketed when its bucketCols is set but its numBucket is not set.
+            // We've seen such table in production.
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java]: Remove this second comment line

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java [hunk 2]
```diff
@@ -41,11 +42,15 @@ public HiveBucketProperty(@JsonProperty("clusteredBy") List<String> clusteredBy,
...
+            return Optional.empty();
         }
         if (!bucketColsSet) {
-            return Optional.empty();
+            throw new PrestoException(
+                    HiveErrorCode.HIVE_INVALID_METADATA,
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java]: Static import this

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java [hunk 3]
```diff
@@ -41,11 +42,15 @@ public HiveBucketProperty(@JsonProperty("clusteredBy") List<String> clusteredBy,
...
+                    format("In metadata of table/partition %s, numBuckets is set to %s, but bucketCol is not set", tablePartitionName, storageDescriptor.getNumBuckets()));
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java]: We can simplify this a bit and put it on one line:
```java
throw new PrestoException(HIVE_INVALID_METADATA, "Table/partition metadata has 'numBuckets' set, but 'bucketCols' is not set: " + tablePartitionName);

Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java]: What does Hive do in this scenario? Does it error or treat as non-bucketed?

**Comment by haozhun** [comment]
Merged #5072

**Pull request closed without merging**